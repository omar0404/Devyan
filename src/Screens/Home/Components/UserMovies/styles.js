import {StyleSheet} from 'react-native';
import theme from 'Devyan/src/Theme';
const styles = StyleSheet.create({
  noMoviesContainer: {
    padding: 18,
    justifyContent: 'center',
    alignItems: 'center',
  },

  noMoviesText: {
    color: theme.text.secondary,
  },
  moviesContainer: {
    marginTop: 10,
  },
});
export default styles;
