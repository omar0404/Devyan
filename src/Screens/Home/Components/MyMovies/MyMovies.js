import React, {Component} from 'react';
import {View, ImageSourcePropType} from 'react-native';
import Header from '../MyMoviesHeader';
import UserMovies from '../UserMovies';
import PropTypes from 'prop-types';
class MyMovies extends Component {
  render() {
    const {movies, onUnLikePress, onMoviePress} = this.props;
    return (
      <View>
        <Header />
        <UserMovies
          movies={movies}
          onMoviePress={onMoviePress}
          onUnLikePress={onUnLikePress}
        />
      </View>
    );
  }
}
MyMovies.propTypes = {
  movies: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.shape({
      id: PropTypes.string,
      posterUri: ImageSourcePropType,
      title: PropTypes.string,
      overview: PropTypes.string,
      date: PropTypes.string,
    }),
  ]),
  onUnLikePress: PropTypes.func,
  onMoviePress: PropTypes.func,
};
export default MyMovies;
