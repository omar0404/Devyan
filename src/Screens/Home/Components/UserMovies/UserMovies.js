import React from 'react';
import {View, Text, ScrollView, ImageSourcePropType} from 'react-native';
import {isEmpty as _isEmpty, map as _map} from 'lodash';
import PropTypes from 'prop-types';
import styles from './styles';
import {MovieCard} from 'Devyan/src/Components';

const renderMovie = (movie, onUnLikePress, onMoviePress) => {
  return (
    <MovieCard
      onPress={() => onMoviePress(movie)}
      onLikePress={() => onUnLikePress(movie)}
      key={movie.id}
      movie={movie}
      isLiked
    />
  );
};
const UserMovies = ({movies, onUnLikePress, onMoviePress}) => {
  const hasMovies = !_isEmpty(movies);
  if (!hasMovies) {
    return (
      <View style={styles.noMoviesContainer}>
        <Text testID={'noMoviesAdded'} style={styles.noMoviesText}>
          No Movies Added
        </Text>
      </View>
    );
  }
  return (
    <ScrollView style={styles.moviesContainer} horizontal>
      {_map(movies, (movie) => renderMovie(movie, onUnLikePress, onMoviePress))}
    </ScrollView>
  );
};
UserMovies.propTypes = {
  onUnLikePress: PropTypes.func,
  onMoviePress: PropTypes.func,
  movies: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.shape({
      id: PropTypes.string,
      posterUri: ImageSourcePropType,
      title: PropTypes.string,
      overview: PropTypes.string,
      date: PropTypes.string,
    }),
  ]),
};
export default UserMovies;
