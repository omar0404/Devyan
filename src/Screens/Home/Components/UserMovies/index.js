import UserMovies from './UserMovies';
export default UserMovies;
