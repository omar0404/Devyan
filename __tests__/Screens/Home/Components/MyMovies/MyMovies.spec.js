import React from 'react';
import MyMovies from 'Devyan/src/Screens/Home/Components/MyMovies';

import {create} from 'react-test-renderer';
import UserMovies from 'Devyan/src/Screens/Home/Components/UserMovies';

describe('MyMovies Componennt', () => {
  it('should render correctly', () => {
    let wrapper = create(<MyMovies />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onMoviePress CB Prop', () => {
    const onMoviePress = jest.fn();
    let wrapper = create(<MyMovies onMoviePress={onMoviePress} />);
    wrapper.root.find(UserMovies).props.onMoviePress();
    expect(onMoviePress).toBeCalled();
  });
});
