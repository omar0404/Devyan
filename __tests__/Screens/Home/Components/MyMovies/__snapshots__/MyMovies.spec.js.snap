// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMovies Componennt should render correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": "auto",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#991F25",
            "height": 15,
            "width": 5,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "rgb(255, 255, 255)",
            "fontSize": 18,
            "marginLeft": 8,
          }
        }
      >
        My Movies
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 18,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "gray",
        }
      }
      testID="noMoviesAdded"
    >
      No Movies Added
    </Text>
  </View>
</View>
`;
